src/cascadia/TerminalSettingsModel/TerminalSettingsSerializationHelpers.h

```diff
@@ -108,10 +108,11 @@ JSON_ENUM_MAPPER(::winrt::Microsoft::Terminal::Control::TextAntialiasingMode)
 // - Helper for converting a user-specified closeOnExit value to its corresponding enum
 JSON_ENUM_MAPPER(::winrt::Microsoft::Terminal::Settings::Model::CloseOnExitMode)
 {
-    JSON_MAPPINGS(3) = {
+    JSON_MAPPINGS(4) = {
         pair_type{ "always", ValueType::Always },
         pair_type{ "graceful", ValueType::Graceful },
         pair_type{ "never", ValueType::Never },
+        pair_type{ "gracefulIfLaunchedByTerminal", ValueType::GracefulIfLaunchedByTerminal },

```

Comment:
Thought - should we change `graceful` to mean "graceful if launched by terminal", and then have the new enum be "graceful even for defterm"? (`gracefulAsDefault)

IDK. I feel like we should keep the current default value as "the sensible thing" and add a new setting for "I want to go back to the old behavior (which wasn't as graceful)".

Naming is hard though.

Reply:
we could do...
- convert `graceful` --> `gracefulIfLaunchedByTerminal`
- add `gracefulAlways`

that way we interpret anybody that explicitly uses `graceful` as the "sensible" one?

Reply:
Generally I care a lot about the naming, but I'm a lot more lenient here because I'd rather have a good transition than a good name haha

Reply:
> IDK. I feel like we should keep the current default value as "the sensible thing" and add a new setting for "I want to go back to the old behavior (which wasn't as graceful)".

That makes sense to me, but also feels like something we should discuss with the team (aaaaaand was about to add the discussion label but I see you've already done that)

Reply:
After teams discussion: we will stick to `gracefulIfLaunchedByTerminal` as the default. Reason:

- Current users who have deliberately set the setting to 'graceful' will have that meaning changed under them
- Users who have not deliberately set the setting get the desired 'sensible' behaviour

Reply:
`automatic` is the default